homeassistant/components/modbus/climate.py

```diff
@@ -270,25 +269,7 @@ def _read_register(self, register_type, register) -> Optional[float]:
 
         return register_value
 
-    def _write_register(self, register, value):
+    async def _write_register(self, register, value):
         """Write holding register using the Modbus hub slave."""
-        try:
-            self._hub.write_registers(self._slave, register, [value, 0])
-        except ConnectionException:
-            self._set_unavailable(register)
-            return
-
+        await self._hub.write_registers(self._slave, register, [value, 0])
         self._available = True
-
-    def _set_unavailable(self, register):
```

Comment:
Ideally this retain the logging (since its already in another place, it would be good to centralize this to keep the code DRY) when it goes unavailable per https://developers.home-assistant.io/docs/integration_quality_scale_index/

"Handles device/service unavailable. Log a warning once when unavailable, log once when reconnected."

Reply:
That should be done in the latest version (see my comment on git rebase, which might have caused you to look at not updated code).